Paella 7 custom themes

miesgre opened the pull request:
This PR adds the new paella feature (skins) to Opencast.
With this, each tenant can configure the paella skin/theme (use other colors, or other icon set)

### Your pull request should…

* [x] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

miesgre committed:
refactor the code for better maintainability

miesgre committed:
Paella 7: Skins support added

gregorydlogan commented:
Should this go into `r/14.x`?

miesgre commented:
It would be great. I can redirect this PR to r/14.x.
What do you think @mliradelc ?

mliradelc commented:
I agree with Greg! This should go to 14.x

miesgre committed:
Theme can be configured in config.json file

miesgre commented:
> I agree with Greg! This should go to 14.x

Done!

miesgre committed:
Update opencast plugins to use the paella skinning feature

miesgre committed:
Fix default language to navigator language

karendolan commented on etc/ui-config/mh_default_org/paella7/default_theme/theme.json:
```diff
@@ -0,0 +1,7 @@
+{
+    "styleSheets": [
```
The Opencast UIConfigRest.java has a GET  @Path("{component}/{filename}")
That service can't serve a file at @Path("{component}/{theme}/{filename}"). Should all these paella7 themes be copied into the engage-player-paella-7 or some other static hosting module prior to build via maven?

miesgre replied:
Thanks @karendolan for testing this. 
You are right. I tested it with the proxy and did not detect this problem. I thought that everything in that folder was 
served. Sorry.

I will try to fix it!

miesgre replied:
@karendolan I updated the PR.

miesgre committed:
Allow sub folder in UIConfigRest to be served

miesgre committed:
Fix paths in config.json and add missing files

miesgre committed:
Default opencast theme updated to use skin feature (better maintainability).

miesgre commented:
@mliradelc, @karendolan Can you review it?

miesgre committed:
update default theme

mliradelc commented on docs/guides/admin/docs/modules/paella.player7/configuration.md:
````diff
@@ -28,10 +28,58 @@ The configurations for the player are done for each tenant. So the configuration
+}
+```
+
+Default theme: `.../etc/ui-config/<tenant>/paella7/default_theme/theme.json`.
````
There is no "default_theme" folder.

miesgre replied:
Documentation updated!

mliradelc commented on etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json:
```diff
@@ -0,0 +1,7 @@
+{
+    "styleSheets": [
+        "theme.css"
```
There is a folder called `custom_theme_example` but also (Maybe left from another PR) `custom_theme.css`

miesgre replied:
`custom_theme_example` is just an example. To use it you need to change in the config.json file the `opencast.theme` property to `custom_theme_example`.

`

mliradelc replied:
Now I checked between the compiled binary and the source code. In the binary, there is an additional file
 `custom_theme.css` that is outside the folder.
 
![image](https://github.com/opencast/opencast/assets/8040628/8fb75ef3-7a10-4ead-94eb-703f53520861)

If the custom themes should be only in folders, what is that `CSS` file?

miesgre replied:
An old file. make a clean. That file should not be there.

mliradelc replied:
Strange, I did a `mvn  clean install.` I will reset the folder and then a new build

mliradelc replied:
> An old file. make a clean. That file should not be there.

That was the problem `git clean -dfx` solved the issue.

mliradelc reviewed (changes requested):
I tried the skin PR, and I don't know why, but I changed several colors from the theme, and nothing changed. I tried with `custom_theme_example/theme.css` and `custom_theme.css`, but no one affects the colors from the player.

I tried also disabling the browser cache without avail.

stefanosgeo commented:
@miesgre  do you have time today to check for @mliradelc  comments? We are going to release Opencast 14.x in about 6 hours.

miesgre committed:
Update paella 7 documentation

miesgre commented:
> I tried the skin PR, and I don't know why, but I changed several colors from the theme, and nothing changed. I tried with `custom_theme_example/theme.css` and `custom_theme.css`, but no one affects the colors from the player.
> 
> I tried also disabling the browser cache without avail.


@stefanosgeo, @mliradelc, I updated the documentation.

To test it you need a to create a `custom theme`. As an example there is a `basic theme` called `custom_theme_example`.

Rename the folder to `custom_example` or change the config.json file the `opencast.theme` property to the folder name.
You can change the colors or icons to test it.

mliradelc commented:
Thanks for the changes.

I'm Testing right now :)

miesgre commented:
@mliradelc You can find the `default opencast theme` here: https://github.com/opencast/opencast/tree/9294dbb6cf966695d5e4cb7a9dbb406347b83a20/modules/engage-paella-player-7/public/default_theme
but it is embedded in the paella7 bundle. It is used when no custom theme is defined or there is an error loading the resources.

mliradelc commented:
> @mliradelc You can find the `default opencast theme` here: https://github.com/opencast/opencast/tree/9294dbb6cf966695d5e4cb7a9dbb406347b83a20/modules/engage-paella-player-7/public/default_theme but it is embedded in the paella7 bundle. It is used when no custom theme is defined or there is an error loading the resources.

I supposed that the default theme was in the etc files, but is inside the jar bundle!

mliradelc reviewed (approved):
Now I tested, and the themes changed. I can change the icons and some CSS colors. The browser loads the `CSS` file.

The additional `CSS` file doesn't appear anymore. That surprised me because a c"lean install"  was not enough.

One more thing,
I could not change the playback bar's color and couldn't find the CSS parameter to modify the color for that zone. 

I approve it because it works as expected.
